code/modules/mob/living/silicon/ai/ai_say.dm

```diff
@@ -49,108 +49,137 @@
 
 	if(!message || announcing_vox > world.time)
 		return
 
-	last_announcement = message
-
 	if(incapacitated())
 		return
 
 	if(control_disabled)
 		to_chat(src, "<span class='warning'>Wireless interface disabled, unable to interact with announcement PA.</span>")
 		return
 
-	var/list/words = splittext(trim(message), " ")
-	var/list/incorrect_words = list()
-
-	if(words.len > 30)
-		words.len = 30
-
-	for(var/word in words)
-		word = lowertext(trim(word))
-		if(!word)
-			words -= word
-			continue
-		if(!GLOB.vox_sounds[word])
-			incorrect_words += word
-
-	if(incorrect_words.len)
-		to_chat(src, "<span class='notice'>These words are not available on the announcement system: [english_list(incorrect_words)].</span>")
+	if(length(message) > max_characters)
+		to_chat(src, "<span class='notice'>You have too many characters! You used [length(message)] characters, you need to lower this to [max_characters] or lower.</span>")
+		return
+	var/regex/check_for_bad_chars = regex("\[^a-zA-Z!?.,' :\]+")
+	if(check_for_bad_chars.Find(message))
+		to_chat(src, "<span class='notice'>These characters are not available on the 15.ai system: [english_list(check_for_bad_chars.group)].</span>")
 		return
+	last_announcement = message
 
 	announcing_vox = world.time + VOX_DELAY
 
-	log_game("[key_name(src)] made a vocal announcement with the following message: [message].")
-	log_talk(message, LOG_SAY, tag="VOX Announcement")
-
-	for(var/word in words)
-		play_vox_word(word, src.z, null)
-
-
-/proc/play_vox_word(word, z_level, mob/only_listener)
-
-	word = lowertext(word)
-
-	if(GLOB.vox_sounds[word])
-
-		var/sound_file = GLOB.vox_sounds[word]
-		var/sound/voice = sound(sound_file, wait = 1, channel = CHANNEL_VOX)
-		voice.status = SOUND_STREAM
-
-	// If there is no single listener, broadcast to everyone in the same z level
+	log_game("[key_name(src)] started making a 15.AI announcement with the following message: [message]")
+	message_admins("[key_name(src)] started making a 15.AI announcement with the following message: [message]")
+	play_vox_word(message, character_to_use, src, src.z, null)
+
+
+/proc/play_vox_word(message, character, mob/living/silicon/ai/speaker, z_level, mob/only_listener)
+	var/api_url = "https://api.15.ai/app/getAudioFile"
+	var/static/vox_voice_number = 0
+	var/datum/http_request/req = new()
+	vox_voice_number++
+	req.prepare(RUSTG_HTTP_METHOD_POST, api_url, "{\"character\":\"[character]\",\"text\":\"[message]\",\"emotion\":\"Contextual\"}", list("Content-Type" = "application/json", "User-Agent" = "/tg/station 13 server"), json_encode(list("output_filename" = "data/vox_[vox_voice_number].wav")))
+	req.begin_async()
+	UNTIL(req.is_complete())
+	var/datum/http_response/res = req.into_response()
+	if(res.status_code == 200)
+		var/full_name_file = "data/vox_[vox_voice_number].wav"
```

Comment:
These should be stored in in tmp, not data, so that way tgs cleans them up every compile if something (like a crash) keeps that from happening.

Also, you could change this to `base_file_path` that contains `data/vox_[vox_voice_number]` so you can just do `[base_file_path].wav` or `[base_file_path].mp3`